virtual_rainforest/core/config.py

```diff
@@ -425,9 +436,20 @@ def validate_with_defaults(
         ValidatorWithDefaults(comb_schema, format_checker=FormatChecker()).validate(
             config_dict
         )
-    except exceptions.ValidationError as err:
+    except exceptions.ValidationError:
+        # Find full set of errors
+        errors = ValidatorWithDefaults(
+            comb_schema, format_checker=FormatChecker()
+        ).iter_errors(config_dict)
+        # Then log all errors in validating core config
+        for error in errors:
+            # Construct details of the tag associated with the error
+            tag = ""
+            for k in error.path:
+                tag += f"[{k}]"
+            LOGGER.error("%s: %s" % (tag, error.message))
         log_and_raise(
-            f"Validation of configuration files failed: {err.message}",
+            "Validation of configuration files failed see above errors",
             ConfigurationError,
         )
```

Comment:
Given that this is done identically above - and it is more substantial that just raising an exception - could we put it inside a function, so it is easier to test independently and reuse, if needed, in other places?